spv/rescan.go

```diff
@@ -55,12 +55,12 @@ func (s *Syncer) rescanCheckTransactions(matches *[]*wire.MsgTx, fadded *blockcf
 		for i, output := range tx.TxOut {
 			_, addrs, _, err := txscript.ExtractPkScriptAddrs(
 				output.Version, output.PkScript,
-				s.wallet.ChainParams())
+				s.chainParams)
```

Comment:
Instead of having the chainparams saved as a property of the `Syncer` struct, why not read it from the current wallet using the walletID property passed into this method?
```suggestion
				s.wallets[walletID].ChainParams())
```